File: app/src/main/java/tr/name/fatihdogan/books/api/response/IndustryIdentifiers.java
package tr.name.fatihdogan.books.api.response;

import android.support.annotation.Keep;
import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Keep
@SuppressWarnings({"unused", "WeakerAccess"})
public class IndustryIdentifiers {

    //region IndustryIdentifier
    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
            INDUSTRY_IDENTIFIER_ISBN_10,
            INDUSTRY_IDENTIFIER_ISBN_13,
            INDUSTRY_IDENTIFIER_ISSN,
            INDUSTRY_IDENTIFIER_OTHER,

    })
    public @interface IndustryIdentifier {

    }

    public static final String INDUSTRY_IDENTIFIER_ISBN_10 = "ISBN_10";
    public static final String INDUSTRY_IDENTIFIER_ISBN_13 = "ISBN_13";
    public static final String INDUSTRY_IDENTIFIER_ISSN = "ISSN";
    public static final String INDUSTRY_IDENTIFIER_OTHER = "OTHER";
    //endregion

    @IndustryIdentifier
    public String type;
    public String identifier;
}

File: app/src/main/java/com/fthdgn/books/api/response/IndustryIdentifiers.kt
@file:Suppress("unused")

package com.fthdgn.books.api.response

import android.support.annotation.Keep
import android.support.annotation.StringDef

@Keep
class IndustryIdentifiers {

    @IndustryIdentifier
    var type: String? = null
    var identifier: String? = null

    companion object {
        @Retention(AnnotationRetention.SOURCE)
        @StringDef(INDUSTRY_IDENTIFIER_ISBN_10, INDUSTRY_IDENTIFIER_ISBN_13, INDUSTRY_IDENTIFIER_ISSN, INDUSTRY_IDENTIFIER_OTHER)
        annotation class IndustryIdentifier

        const val INDUSTRY_IDENTIFIER_ISBN_10 = "ISBN_10"
        const val INDUSTRY_IDENTIFIER_ISBN_13 = "ISBN_13"
        const val INDUSTRY_IDENTIFIER_ISSN = "ISSN"
        const val INDUSTRY_IDENTIFIER_OTHER = "OTHER"
    }
}
